tests/Microsoft.Identity.Test.Unit/CoreTests/WsTrustTests/WsTrustTests.cs

```diff
@@ -129,5 +130,43 @@ public async Task WsTrustRequestParseErrorTestAsync()
                 }
             }
         }
+
+        [TestMethod]
+        [Description("WsTrustRequest encounters a non parseable response from the wsTrust endpoint")]
+        public async Task WsTrustRequestTokenNotFoundInResponseTestAsync()
+        {
+            const string uri = "https://some/address/usernamemixed";
+
+            var endpoint = new WsTrustEndpoint(new Uri(uri), WsTrustVersion.WsTrust2005);
+
+            using (var harness = CreateTestHarness())
+            {
+                harness.HttpManager.AddMockHandler(
+                    new MockHttpMessageHandler()
+                    {
+                        ExpectedUrl = uri,
+                        ExpectedMethod = HttpMethod.Post,
+                        ResponseMessage = new HttpResponseMessage(HttpStatusCode.OK)
+                        {
+                            Content = new StringContent(
+                               File.ReadAllText(ResourceHelper.GetTestResourceRelativePath("WsTrustResponseNoToken.xml")))
+                        }
+                    });
+
+                var requestContext = new RequestContext(harness.ServiceBundle, Guid.NewGuid());
+                try
+                {
+                    var message = endpoint.BuildTokenRequestMessageWindowsIntegratedAuth("urn:federation:SomeAudience");
+
+                    WsTrustResponse wstResponse =
```

Comment:
nit: Use `AssertException.TaskThrowsAsync` like in: https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/2f83c8e783048734cdec26c4358c96d8f30a0b78/tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/RuntimeBrokerTests.cs#L263-L268